FirebaseAdmin/FirebaseAdmin.IntegrationTests/OidcProviderConfigTest.cs

```diff
@@ -0,0 +1,172 @@
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FirebaseAdmin.Auth;
+using FirebaseAdmin.Auth.Providers;
+using Xunit;
+
+namespace FirebaseAdmin.IntegrationTests
+{
+    [TestCaseOrderer(
+        "FirebaseAdmin.IntegrationTests.TestRankOrderer", "FirebaseAdmin.IntegrationTests")]
+    public class OidcProviderConfigTest : IClassFixture<OidcProviderConfigFixture>
+    {
+        private readonly OidcProviderConfigFixture fixture;
+
+        public OidcProviderConfigTest(OidcProviderConfigFixture fixture)
+        {
+            this.fixture = fixture;
+        }
+
+        [Fact]
+        [TestRank(0)]
+        public void CreateProviderConfig()
+        {
+            var config = this.fixture.ProviderConfig;
+
+            Assert.Equal("OIDC_DISPLAY_NAME", config.DisplayName);
+            Assert.True(config.Enabled);
+            Assert.Equal("OIDC_CLIENT_ID", config.ClientId);
+            Assert.Equal("https://oidc.com/issuer", config.Issuer);
+        }
+
+        [Fact]
+        [TestRank(10)]
+        public async Task GetProviderConfig()
+        {
+            var config = await FirebaseAuth.DefaultInstance.GetOidcProviderConfigAsync(
+                this.fixture.ProviderId);
+
+            Assert.Equal("OIDC_DISPLAY_NAME", config.DisplayName);
+            Assert.True(config.Enabled);
+            Assert.Equal("OIDC_CLIENT_ID", config.ClientId);
+            Assert.Equal("https://oidc.com/issuer", config.Issuer);
+        }
+
+        [Fact]
+        [TestRank(10)]
+        public async Task ListProviderConfig()
+        {
+            OidcProviderConfig config = null;
+
+            var pagedEnumerable = FirebaseAuth.DefaultInstance.ListOidcProviderConfigsAsync(null);
+            var enumerator = pagedEnumerable.GetEnumerator();
+            while (await enumerator.MoveNext())
+            {
+                if (enumerator.Current.ProviderId == this.fixture.ProviderId)
+                {
+                    config = enumerator.Current;
+                    break;
+                }
+            }
+
+            Assert.NotNull(config);
+            Assert.Equal("OIDC_DISPLAY_NAME", config.DisplayName);
+            Assert.True(config.Enabled);
+            Assert.Equal("OIDC_CLIENT_ID", config.ClientId);
+            Assert.Equal("https://oidc.com/issuer", config.Issuer);
+        }
+
+        [Fact]
+        [TestRank(20)]
+        public async Task UpdateProviderConfig()
+        {
+            var args = new OidcProviderConfigArgs()
+            {
+                ProviderId = this.fixture.ProviderId,
+                DisplayName = "UPDATED_OIDC_DISPLAY_NAME",
+                Enabled = false,
+                ClientId = "UPDATED_OIDC_CLIENT_ID",
+                Issuer = "https://oidc.com/updated-issuer",
+            };
+
+            var config = await FirebaseAuth.DefaultInstance.UpdateProviderConfigAsync(args);
+
```

Comment:
Can you assert the value of `ProviderId` here because it's deterministic (i.e., `this.fixture.ProviderId`)?

Reply:
Done